eng/WpfArcadeSdk/tools/InjectModuleInitializer/AddModuleConstructorTask.cs

```diff
@@ -41,31 +41,33 @@ public override bool Execute()
                 var moduleConstructorIL = File.ReadAllText(ModuleConstructorIL);
                 var msCorLibAssemblySectionIL = File.ReadAllText(MsCorLibAssemblySectionIL);
                 var sourceIL = File.ReadAllText(ILFile);
+                string ilToInject = msCorLibAssemblySectionIL + Environment.NewLine + moduleConstructorIL;
 
                 using (var outputIL = new StreamWriter(File.Open(ILFile, FileMode.Truncate)))
                 {
-                    if (Regex.Match(sourceIL, @"\.assembly extern mscorlib", RegexOptions.Singleline) == Match.Empty)
-                    {
-                        outputIL.WriteLine(msCorLibAssemblySectionIL);
-                    }
-
                     if (Regex.Match(sourceIL, @"\.class.+?ModuleInitializer.+?\.method.+?static.+?Initialize\(\).+?end of class ModuleInitializer", RegexOptions.Singleline) == Match.Empty)
                     {
                         Log.LogError("Inserting a module initializer requires the assembly to implement a class named ModuleInitializer with a static parameterless method named Initialize.");
                         return false;
                     }
 
-                    if (Regex.Match(sourceIL, @"\.class private auto ansi '\<Module\>'.+?\.method private hidebysig specialname rtspecialname static void \.cctor \(\) cil managed ", RegexOptions.Singleline) == Match.Empty)
-                    {
-                        outputIL.WriteLine(moduleConstructorIL);
-                    }
-                    else
+                    if (Regex.Match(sourceIL, @"\.class private auto ansi '\<Module\>'.+?\.method private hidebysig specialname rtspecialname static void \.cctor \(\) cil managed ", RegexOptions.Singleline) != Match.Empty)
                     {
                         Log.LogError("Cannot insert a module initializer into an assembly that already contains one.");
                         return false;
                     }
 
-                    outputIL.WriteLine(sourceIL);
+                    // Attempt to replace an existing mscorlib reference with the reference + module initializer.
+                    string modifiedIL = Regex.Replace(sourceIL, @"\.assembly extern mscorlib.+};", ilToInject, RegexOptions.Singleline);

```

Comment:
You're now looking for a multi-line pattern, from ".assembly extern mscorlib" to the matching "};" several lines later.
So SingleLine isn't right.

Reply:
I take that back.  "Singleline" does not mean what I thought it meant.  Never start a land war in Asia.

Reply:
That's not correct, there are multiple multi-line patterns in the checks, all of which use SingleLine.

SingleLine means it match across newline characters, see: https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regexoptions?view=netframework-4.8

It's counter-intuitive (at least to me).